react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { counter: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    const value = this.state.counter + 1;
+    this.setState({ counter: value });
+  }
+
+  render() {
+    if (this.state.counter < 3) {
+      return <button onClick={this.handleClick} className='zero' >Hot Button</button>;
+    } else if (this.state.counter < 6) {
+      return <button onClick={this.handleClick} className='three' >Hot Button</button>;
+    } else if (this.state.counter < 9) {
+      return <button onClick={this.handleClick} className='six'> Hot Button</button>;
+    } else if (this.state.counter < 12) {
+      return <button onClick={this.handleClick} className='nine' >Hot Button</button>;
+    } else if (this.state.counter < 15) {
+      return <button onClick={this.handleClick} className='twelve'> Hot Button</button>;
+    } else if (this.state.counter < 18) {
+      return <button onClick={this.handleClick} className='fifteen'> Hot Button</button>;
+    } else {
+      return <button onClick={this.handleClick} className='eighteen'> Hot Button</button>;
+    }
+  }
+}
+export default <Button />;
```

Comment:
Here we want to export the class and not a React element, otherwise, we can't have more than one of them in our program.

Reply:
🤔 Actually that isn't true. However, we typically don't export react elements, we export components (functions or classes). I'm not actually sure if there's a specific reason except it makes putting components together really weird.

```jsx
<div>
  <Button />
</div>

// vs.

<div>
  { Button }
</div>
```